test/test_data_reader.rb

```diff
@@ -156,4 +183,14 @@ def ignore_warnings
       $-w = old_verbose
     end
   end
+
+  def capture_stderr
+    old_stderr = $stderr
+    $stderr = StringIO.new
+    begin
+      yield
```

Comment:
Could `capture_stderr` return `$stderr.string` so tests can verify is something was written to stderr or not?